spynnaker/pyNN/models/neural_projections/connectors/kernel_connector.py

```diff
@@ -232,11 +232,21 @@ def __get_kernel_vals(self, vals):
                 vals.shape[HEIGHT] == self._kernel_h and
                 vals.shape[WIDTH] == self._kernel_w):
             return vals.view(ConvolutionKernel)
-        # TODO: make this error more descriptive?
+        elif ((isinstance(vals, list))):
+            height = len(vals)
+            if height == self._kernel_h:
+                correct_size = True
+                for h in range(height):
+                    width = len(vals[h])
+                    if width != self._kernel_w:
+                        correct_size = False
+                if correct_size:
+                    return numpy.asarray(vals).view(ConvolutionKernel)
```

Comment:
Would it be easier to add a check for a list in advance of this and then do vals = numpy.asarray(vals) there?  Then the other checks would happen automatically?

Reply:
Agreed - done.